source/geod24/concurrency.d

```diff
@@ -1501,6 +1501,176 @@ private:
     assert(x[0] == 5);
 }
 
+/***********************************************************************
+
+    A common interface for objects that can be used in `select()`
+
+***********************************************************************/
+
+public interface Selectable
+{
+    /***********************************************************************
+
+        Try to read/write to the `Selectable` without blocking. If the
+        operation would block, queue and link it with the `sel_state`
+
+        Params:
+            ptr = pointer to the data for the select operation
+            sel_state = SelectState instace of the select call being executed
+            sel_id = id of the select call being executed
+
+    ***********************************************************************/
+
+    void selectWrite (void* ptr, SelectState sel_state, int sel_id);
+
+    /// Ditto
+    void selectRead (void* ptr, SelectState sel_state, int sel_id);
+}
+
+/***********************************************************************
+
+    An aggregate to hold neccessary information for a select operation
+
+***********************************************************************/
+
+public struct SelectEntry
+{
+    /// Reference to a Selectable object
+    Selectable selectable;
+
+    /// Pointer to the select data
+    void* select_data;
+
+    /***********************************************************************
+
+        Default ctor
+
+        Params:
+            selectable = A selectable interface reference
+            select_data = pointer to the data for the select operation
+
+    ***********************************************************************/
+
+    this (Selectable selectable, void* select_data)
+    {
+        this.selectable = selectable;
+        this.select_data = select_data;
+    }
```

Comment:
D gives you a default ctor for struct. So I assume you only want it to be constructed with two parameter ?

Reply:
This was my thinking too. It prevents building a partially constructed aggregate. Kind of like `explicit` in C++.

Reply:
Yes exactly, both parameters are required.